lib/immutable-struct.rb

```diff
@@ -83,7 +83,9 @@ def self.new(*attributes,&block)
             attr_value = attrs[ivar_name.to_s].nil? ? attrs[ivar_name.to_sym] : attrs[ivar_name.to_s]
             instance_variable_set("@#{ivar_name}", attr_value)
           end
+          instance_variable_get("@#{ivar_name}").public_send(:freeze)
```

Comment:
Coming back to this PR (sorry for the delay.

I'm not sure about this line.  `BasicObject` doesn't support this and I'm not sure we aren't using them anywhere.  Also worried that this could freeze an object that is shared elsewhere. This also won't freeze stuff inside a collection, giving a false sense of security.

I wonder if we should move this to a `freeze` method so that anyone that wants frozen objects can call it and deal with the consequences on an opt-in basis?